### PR TITLE
Remove legacy scaffolding method and drop support for `v1alpha1` templates

### PR DESCRIPTION
**Pull request by benjdlambert**
Signed-off-by: blam <ben@blam.sh>

Here's the cleaning of the `scaffolder` and `catalog` of the `v1alpha` template support.

For merge after 1st of June release to drop support for the `alpha` Templates and syntax.

#### :heavy_check_mark: Checklist

- [ ] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [ ] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: plugins/catalog-backend/src/ingestion/LocationReaders.ts [hunk 1]
```diff
@@ -265,6 +265,7 @@ export class LocationReaders implements LocationReader {
           )}, ${e}`;
           emit(result.inputError(item.location, message));
           logger.warn(message);
+          logger.debug(e.stack);
```
Comment (jhaals) [inline plugins/catalog-backend/src/ingestion/LocationReaders.ts:268]: is this related?

File: plugins/catalog-backend/src/next/processing/DefaultCatalogProcessingOrchestrator.ts [hunk 2]
```diff
@@ -124,6 +124,7 @@ export class DefaultCatalogProcessingOrchestrator
       };
     } catch (error) {
       this.options.logger.warn(error.message);
+      this.options.logger.debug(error.stack);
```
Comment (jhaals) [inline plugins/catalog-backend/src/next/processing/DefaultCatalogProcessingOrchestrator.ts:127]: nice to have, separate pr 👍

File: plugins/scaffolder-backend/src/service/router.test.ts [hunk 3]
```diff
@@ -33,12 +33,13 @@ import {
...
   DatabaseManager,
   UrlReaders,
+  DockerContainerRunner,
 } from '@backstage/backend-common';
 import { CatalogApi } from '@backstage/catalog-client';
 import { ConfigReader } from '@backstage/config';
 import express from 'express';
 import request from 'supertest';
-import { Preparers, Publishers, Templaters } from '../scaffolder';
+import { TemplateEntityV1beta2 } from '../../../../packages/catalog-model/src';
```
Comment (freben) [inline plugins/scaffolder-backend/src/service/router.test.ts:42]: oops, cross-package import

File: plugins/catalog-backend/src/ingestion/LocationReaders.ts (hunk 1 above)
Comment (freben) [reply plugins/catalog-backend/src/ingestion/LocationReaders.ts:268]: nah remove

File: plugins/scaffolder-backend/src/scaffolder/actions/builtin/fetch/cookiecutter.ts [hunk 4]
```diff
@@ -14,22 +14,116 @@
...
 import fs from 'fs-extra';
-import { resolve as resolvePath } from 'path';
-import { TemplaterBuilder, TemplaterValues } from '../../../stages/templater';
+import path, { resolve as resolvePath } from 'path';
+import { Writable } from 'stream';
+import { runCommand } from '../helpers';
 import { createTemplateAction } from '../../createTemplateAction';
 import { fetchContents } from './helpers';
 
+export class CookiecutterRunner {
```
Comment (freben) [inline plugins/scaffolder-backend/src/scaffolder/actions/builtin/fetch/cookiecutter.ts:33]: you deleted the tests, are these covered elsewhere?

**Review by freben — COMMENTED** [review]
This needs changesets, particularly you probably want to update create-app including migration instructions

File: plugins/scaffolder-backend/src/scaffolder/actions/builtin/fetch/cookiecutter.test.ts [hunk 5]
```diff
@@ -65,122 +70,147 @@ describe('fetch:cookiecutter', () => {
...
+      baseUrl: 'somebase',
+      workspacePath: mockTmpDir,
+      logger: getVoidLogger(),
+      logStream: new PassThrough(),
+      output: jest.fn(),
+      createTemporaryDirectory: jest.fn().mockResolvedValue(mockTmpDir),
+    };
+
+    // mock the temp directory
+    mock({ [mockTmpDir]: {} });
```
Comment (freben) [inline plugins/scaffolder-backend/src/scaffolder/actions/builtin/fetch/cookiecutter.test.ts:97]: dude we really should name this import not just `mock` :) `mockFs` seems good to me

File: plugins/scaffolder-backend/src/scaffolder/actions/builtin/fetch/cookiecutter.test.ts [hunk 6]
```diff
@@ -65,122 +70,147 @@ describe('fetch:cookiecutter', () => {
...
-    await action.handler(mockContext);
-
-    expect(cookiecutterTemplater.run).toHaveBeenCalledWith({
-      workspacePath: mockTmpDir,
-      logStream: mockContext.logStream,
-      values: mockContext.input.values,
-    });
+    await expect(
+      action.handler(mockContext),
+    ).rejects.toThrowErrorMatchingInlineSnapshot(
```
Comment (freben) [inline plugins/scaffolder-backend/src/scaffolder/actions/builtin/fetch/cookiecutter.test.ts:130]: why not just `toThrow`? 🤔

**Review by freben — APPROVED** [review]
Just don't forget the changesets!

File: plugins/scaffolder-backend/src/scaffolder/actions/builtin/fetch/cookiecutter.test.ts (hunk 6 above)
Comment (benjdlambert) [reply plugins/scaffolder-backend/src/scaffolder/actions/builtin/fetch/cookiecutter.test.ts:130]: It was just a pattern I saw that's pretty nice for when you change error messages it makes updating the tests easier because they're snapshots. :shrug: Can change.

File: plugins/scaffolder-backend/src/scaffolder/actions/builtin/fetch/cookiecutter.test.ts (hunk 5 above)
Comment (benjdlambert) [reply plugins/scaffolder-backend/src/scaffolder/actions/builtin/fetch/cookiecutter.test.ts:97]: Ha this was left over from the previous tests.. good call.

File: plugins/scaffolder-backend/src/scaffolder/actions/builtin/fetch/cookiecutter.test.ts (hunk 6 above)
Comment (freben) [reply plugins/scaffolder-backend/src/scaffolder/actions/builtin/fetch/cookiecutter.test.ts:130]: Nah just curious
